src/addProp.ts

```diff
@@ -1,8 +1,15 @@
+import type { UpsertProp } from "./internal/types";
 import { purry } from "./purry";
 
 /**
  * Add a new property to an object.
  *
+ * The function doesn't do any checks on the input object. If the property
+ * already exists it will be overwritten, and the type of the new value is not
+ * checked against the previous type.
+ *
+ * Use `set` to override values explicitly with better protections.
+ *
```

Comment:
**thought:** i now wonder if we can add an equivalent to `setPath`, which would fulfill #554, called `addPath`?

Reply:
I think we can overhaul the whole set of functions that replace Lodash's `set` function. All these changes came about while I was writing the migration for Lodash. Some of the changes might be breaking, so we might need to wait for a v3 (which I hope to release sooner, maybe in six months).

I'm happy with the way the new migration docs are turning out because it's obvious that the simple mapping table hid many complexities, which can now be worked on more clearly.